activities/views.py

```diff
@@ -0,0 +1,105 @@
+        queryset = Screening.objects.get_queryset().order_by('id')
+
+
+        uc_id = request.POST.get('user_created') # POST param 'user_created', default value is empty string
+        if uc_id:
+            queryset = queryset.filter(user_created__exact=uc_id) # filters for numeric values with exact match 
+
+        start_day = request.POST.get('start_day')
+        start_month = request.POST.get('start_month')
+        start_year = request.POST.get('start_year')
+        end_day = request.POST.get('end_day')
+        end_month = request.POST.get('end_month')
+        end_year = request.POST.get('end_year')
+
+        # case1: all values are present
+        if start_day and start_month and start_year and end_day and end_month and end_year:
+            # params type value is string, trimmed spaces,convert to int and then make date by combining values
+            try:
+                start_date = datetime.date(int(start_year.strip()),int(start_month.strip()), int(start_day.strip()))
+                end_date = datetime.date(int(end_year.strip()),int(end_month.strip()), int(end_day.strip()))       
+                queryset = queryset.filter(date__range=(start_date, end_date)) # filters values in date range
+            except:
+                print("Date error occurred")
+        # case2: only start values are present 
+        elif start_day and start_month and start_year and not end_day and not end_month and not end_year:
+            try:
+                start_date = datetime.date(int(start_year.strip()),int(start_month.strip()), int(start_day.strip()))      
+                queryset = queryset.filter(date__gte=start_date) # filters values greater than or equal to start date
+            except:
+                print("Start Date error occurred")
+        # case3: only end values are present 
+        elif not start_day and not start_month and not start_year and  end_day and  end_month and  end_year:
+            try:
+                end_date = datetime.date(int(end_year.strip()),int(end_month.strip()), int(end_day.strip()))       
+                queryset = queryset.filter(date__lte=end_date) # filters values less than or equal to end date
+            except:
+                print("End Date error occurred")
+
+        start_limit = request.POST.get('start_limit') # POST param 'start_limit'
+        end_limit = request.POST.get('end_limit') # POST param 'end_limit'  
+
+        # limits the total response count        
+        if start_limit and end_limit: # case1: both are present
+            queryset = queryset[int(start_limit)-1:int(end_limit)]
+        elif start_limit: # case2: only start_limit is present
+            queryset = queryset[int(start_limit)-1:]
+        elif end_limit: # case3: only end_limit is present
+            queryset = queryset[:int(end_limit)] 
+
+        count = self.request.POST.get("count", "False")
+        if count in ["True","true","t","T","Yes","yes","Y","y"]:
```

Comment:
add ["True","true","t","T","Yes","yes","Y","y"] to a variable and state the use of using it.

Reply:
Code changed